HELP.md

```diff
@@ -0,0 +1,716 @@
+# Contract best practices
+
+## Main structure and persistent collections
+
+The main contract structure is marked with `#[near_bindgen]`. It has to be serializable and deserializable with [Borsh](https://borsh.io).
+
+```rust
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct Contract {
+    pub data: String,
+    pub owner_id: AccountId,
+    pub value: u128,
+}
+```
+
+Every time an external method is called, the entire structure has to be deserialized.
+The serialized contract data is stored in the persistent storage under the key `STATE`.
+
+Change methods (see below) are serializing the main contract structure at the end and stores the new value into the storage.
+
+Persistent collection helps store extra data in the persistent storage outside of main structure.
+NEAR SDK provides the following collections:
+- `Vector` - An iterable implementation of vector.
+- `LookupMap` - An non-iterable implementation of a map.
+- `LookupSet` - An non-iterable implementation of a set.
+- `UnorderedMap` - An iterable implementation of a map.
+- `UnorderedSet` - An iterable implementation of a set.
+- `TreeMap` - An iterable sorted map based on AVL-tree
+- `LazyOption` - An `Option` for a single value.
+
+Every instance of a persistent collection requires a unique storage prefix.
+The prefix is used to generate internal keys to store data in the persistent storage.
+These internal keys has to be unique to avoid collisions (even with key `STATE`).
+
+## Generating unique prefixes for persistent collections
+
+When a contract gets complicated, there may be multiple different
+collections that may not be all part of the main structure, but instead be part of sub-structure or nested collections.
+They all need to have unique prefixes.
+
+#### The traditional way
```

Comment:
should we make this a small stub and link to appendix of doc just to keep the flow and emphasis on "the new way"?